### ISSUE TITLE
test.py问题

### ISSUE THREAD

**Issue by huangjf11** [taishan1994/Chinese-LLaMA-Alpaca-LoRA-Tuning#2]
训练好模型做测试时，不应该针对输入直接做输出吗？为什么preds还要受trues的影响。
        output = model(**batch)
        labels = batch["labels"].detach().cpu().numpy()
        logits = output.logits
        preds = torch.argmax(logits, -1).detach().cpu().numpy()
        preds = preds[:, :-1]
        labels = labels[:, 1:]
        **preds = np.where(labels != -100, preds, tokenizer.pad_token_id)
        decoded_preds = tokenizer.batch_decode(preds, skip_special_tokens=True)
        labels = np.where(labels != -100, labels, tokenizer.pad_token_id)
        decoded_labels = tokenizer.batch_decode(labels, skip_special_tokens=True)**

这段是否存在问题？

**测试时发现preds受trues很强很强的影响**

**Comment by taishan1994** [comment]
这里使用标签里面的哪些是需要输出的信息，并不是使用真正的标签，因为没有提前记录answer的位置。是在chat_ner里面我们是没有label的。

**Comment by lidpeng** [comment]
我也有相似的疑问：
  for s, t in zip(tokenized_sources['input_ids'], tokenized_targets['input_ids']):
      input_ids = (s + t)[:self.max_seq_length]
      labels = ([IGNORE_INDEX] * len(s) + t)[:self.max_seq_length]
      assert len(input_ids) == len(labels)
      input_ids = input_ids + [self.tokenizer.pad_token_id] * (self.max_seq_length - len(input_ids))
      labels = labels + [IGNORE_INDEX] * (self.max_seq_length - len(labels))
在我的测试中发现，在处理数据的时候，input_ids里面加入了预测target的ids，这样相当于是在输入中包含了答案，导致输出很大程度依赖于answer的内容。
